acceptance/tests/direct_puppet/supports_utf8.rb

```diff
@@ -5,14 +5,20 @@
   app_type = File.basename(__FILE__, '.*')
   tmp_environment   = mk_tmp_environment(app_type)
 
-  filenames         = {'windows' => "C:/cygwin64/tmp/#{tmp_environment}_file"}
-  filenames.default = "/tmp/#{tmp_environment}_file"
+  tmp_file = {}
+  agents.each do |agent|
+    osfamily = agent.platform =~ /windows/ ? 'windows' : 'other'
+    # just create one tmpfile per windows and other
+    tmp_file[osfamily] = agent.tmpfile(tmp_environment) unless tmp_file[osfamily]
+  end
   teardown do
     step 'remove the tmp environment symlink' do
       on master, "rm -rf #{File.join(environmentpath, tmp_environment)}"
     end
     step 'clean out produced resources' do
-      on(agents, "rm #{filenames['windows']}; rm #{filenames.default}", :accept_all_exit_codes => true)
+      agents.each_with_index do |agent, index|
+        on(agents, "rm #{tmp_file['windows']}; rm #{tmp_file['other']}", :accept_all_exit_codes => true)
```

Comment:
I see the appeal of doing this in a one-liner, but it does seem mildly dangerous given the paths generated are platform specific.